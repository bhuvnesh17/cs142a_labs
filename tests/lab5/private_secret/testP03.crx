// function with large parameter list

func foo(a:int, b:bool, c:float, d:int, e:bool, f:float) : float {
  if (b and e) {
    return c;
  } else {
    let a = d;
  }
  return c + f / 3.14;
}

func main() : void {
  ::printFloat(::foo(7,true,2.71, 8,false,7.89));
  ::printFloat(::foo(7,true,2, 8,false,7.89));
  ::printFloat(::foo(7,true,2.72, 8,0,7.89));
  ::printFloat(::foo(1,2,3, 4,5,6));
}
