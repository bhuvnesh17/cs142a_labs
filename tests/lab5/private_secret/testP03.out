Error type-checking file.
TypeError(14,16)[Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, bool, int, int, bool, float).]
TypeError(14,3)[Cannot call func(TypeList(float)):void using TypeList(ErrorType(Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, bool, int, int, bool, float).)).]
TypeError(15,16)[Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, bool, float, int, int, float).]
TypeError(15,3)[Cannot call func(TypeList(float)):void using TypeList(ErrorType(Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, bool, float, int, int, float).)).]
TypeError(16,16)[Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, int, int, int, int, int).]
TypeError(16,3)[Cannot call func(TypeList(float)):void using TypeList(ErrorType(Cannot call func(TypeList(int, bool, float, int, bool, float)):float using TypeList(int, int, int, int, int, int).)).]

